lib/protocol/request/index.js

```diff
@@ -91,6 +91,7 @@ function getCommandOptions(options) {
 function authenticate(options) {
   var segment = createSegment(MessageType.AUTHENTICATE, options);
   // authentication
+  segment.add(PartKind.CLIENT_CONTEXT, options.clientContext);
```

Comment:
We normally add this after the authentication request, but I don't think this is really a concern.